src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/StreamMemoryBlockProvider.cs

```diff
@@ -149,20 +154,14 @@ private unsafe bool TryCreateMemoryMappedFileBlock(long start, int size, [NotNul
                 }
             }
 
-            IDisposable accessor = MemoryMapLightUp.CreateViewAccessor(_lazyMemoryMap, start, size);
+            MemoryMappedViewAccessor accessor = _lazyMemoryMap.CreateViewAccessor(start, size, MemoryMappedFileAccess.Read);
```

Comment:
`UnauthorizedAccessException`s used to be wrapped and rethrown inside an `IOException` but now they are thrown directly. Is there a problem with that behavior change?

Reply:
> Is there a problem with that behavior change?

Assuming that exception could escape to calling code, yes, we shouldn't change the type of the exception thrown.

Reply:
Fixed.